### DISCUSSION TITLE
reticulum and yggdrasil

### DISCUSSION THREAD

**Discussion by khimaros** [markqvist/Reticulum#162] [Help & Questions]
is it pragmatic to run reticulum over yggdrasil? my assumption is it would provide some of the benefits of I2P but less anonymity.

it seems like it could be an interesting use case for a "global" `AutoInterface` and would allow transport servers behind NAT.

just curious if this has already been attempted and what caveats might exist.

**Comment by markqvist** [comment]
Yes, that would be totally possible. You could already do it "manually" by setting up UDP or TCP interfaces that target endpoints over Yggdrasil, but the neatest and easiest way would definitely be to have a specific interface type like the current I2P implementation.

> it seems like it could be an interesting use case for a "global" AutoInterface and would allow transport servers behind NAT.

Yes, I agree with that. If Yggdrasil supports global-scope IPv6 multicast routing, it would probably even work "out-of-the-box", since the AutoInterface code already supports settings all the different IPv6 multicast scopes. So in *theory*, if you are connected to a fully functional IPv6 network, you can set the discovery scope to global on the AutoInterface, set your group ID, and it will "just work", globally. But I have yet to actually try this somewhere where global IPv6 multicast routing works. If it does on Yggdrasil, it would be neat.

**Reply by khimaros** [reply]
i did some poking around re: global-scope multicast in yggdrasil repo and couldn't find an answer.

i asked here: https://github.com/yggdrasil-network/yggdrasil-go/discussions/991 and will follow up.

**Comment by samuk** [comment]
Would https://hypercore-protocol.org/ help here?

**Comment by resession** [comment]
@markqvist saw your question in that discussion thread about a dht. yggdrasil uses a dht, https://yggdrasil-network.github.io/about.html also, https://github.com/yggdrasil-network/yggdrasil-go/issues/418 it talks about multicast.

**Comment by resession** [comment]
there is also https://github.com/cjdelisle/cjdns it is a ipv6 network just like yggdrasil

**Reply by khimaros** [reply]
cjdns is the spiritual (and somewhat literal) predecessor to yggdrasil. many of not all of the devs moved their efforts to yggdrasil.

**Reply by marek22k** [reply]
That's right. Except for updating some dependencies, cjdns seems to be dead. There are no answers or progress on many issues.

**Comment by marek22k** [comment]
It would also be interesting to see if it works the other way around: Yggdrasil-over-Reticulum. With I2P, for example, both are possible: I2P-over-Yggdrasil (i2pd implementation) and Yggdrasil-over-I2P (via SOCKS Proxy).
